docs/pages/deploy-a-cluster/teleport-enterprise/gcp-kms.mdx

```diff
@@ -0,0 +1,270 @@
+---
+title: Google KMS
+description: Configure Teleport to store CA private keys in the Google Cloud Key Management Service
+h1: Store Teleport Private Keys in Google Cloud KMS
+---
+
+This guide will show you how to set up your Teleport Cluster to use the Google
+Cloud Key Management Service (KMS) to store and handle the CA private key
+material used to sign all certificates issued by your Teleport cluster.
+
+Teleport generates User, Host, JWT, and Database Certificate Authorities (CAs)
+during the first Auth Server's initial startup.
+These CAs are used to sign all certificates issued to clients and hosts in the
+Teleport cluster.
+When configured to use Google KMS, all private key material for these CAs will
+be generated, stored, and used for signing inside of Google KMS.
+In short, private key material will never leave Google KMS.
+Instead of the actual private key, Teleport will only store the ID of the KMS
+key.
+
+If launching a new Teleport cluster this will all be handled during initial
+startup with no specific interventions required after configuration. If
+configuring an existing Teleport cluster to switch to KMS keys after it has
+already started with software keys a CA rotation must be performed, read on to
+[migrating an existing cluster](#migrating-an-existing-cluster) to learn more.
+
+<ScopedBlock scope={["oss", "cloud"]}>
+
+This guide is intended for self-hosted Teleport Enterprise users.
+
+</ScopedBlock>
+
+## Prerequisites
+
+- Teleport v(=teleport.version=) Enterprise (self-hosted).
+
+- The `tctl` administration tool version >= (=teleport.version=).
+
+  (!docs/pages/includes/tctl.mdx!)
+
+- A Google Cloud account.
+
+<VarList/>
+
+## Step 1/5. Create a key ring in GCP
+
+Each Teleport Auth Server will need to be configured to use a GCP key ring
+which will hold all keys generated and used by that Auth Server.
+If running a High-Availability Teleport cluster with two or more Auth Servers,
+every Auth Server can be configured to use the same key ring, or if desired each
+can be configured to use a unique key ring in a different region (for redundancy
+or to decrease latency).
+
+It is recommended to create a unique key ring for use by Teleport to logically
```

Comment:
```suggestion
It is recommended to create a dedicated key ring for use by Teleport to logically
```

By reusing the word 'unique' from the prior sentence, it sounds like you're recommending a keyring per auth server, but I don't think you are. Here you're just saying to use a keyring dedicated to Teleport, irregardless of whether you use a key ring per region or a single one.